src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyAndFieldNames.Fixer.cs

```diff
@@ -0,0 +1,70 @@
+using Microsoft.CodeAnalysis.Rename;
+using Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    /// <summary>
+    /// CA1071: Constructor parameters should match property and field names.
+    /// Based on <see cref="ParameterNamesShouldMatchBaseDeclarationFixer"/>.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    [Shared]
+    public sealed class ConstructorParametersShouldMatchPropertyAndFieldNamesFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(ConstructorParametersShouldMatchPropertyAndFieldNamesAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxNode syntaxRoot = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            SemanticModel semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+            foreach (Diagnostic diagnostic in context.Diagnostics)
+            {
+                SyntaxNode node = syntaxRoot.FindNode(context.Span);
+                ISymbol declaredSymbol = semanticModel.GetDeclaredSymbol(node, context.CancellationToken);
+
+                if (declaredSymbol.Kind != SymbolKind.Parameter)
+                {
+                    continue;
+                }
+
+                // This approach is very naive. Most likely we want to support NamingStyleOptions, available in Roslyn.
+                string newName = LowerFirstLetter(diagnostic.Properties[ConstructorParametersShouldMatchPropertyAndFieldNamesAnalyzer.ReferencedFieldOrPropertyName]);
+
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        string.Format(CultureInfo.CurrentCulture, MicrosoftNetCoreAnalyzersResources.ConstructorParametersShouldMatchPropertyNamesTitle, newName),
+                        cancellationToken => GetUpdatedDocumentForParameterRenameAsync(context.Document, declaredSymbol, newName, cancellationToken),
+                        nameof(ConstructorParametersShouldMatchPropertyAndFieldNamesFixer)),
+                    diagnostic);
+            }
+        }
+
+        private static string LowerFirstLetter(string targetName)
+        {
+            return $"{targetName[0].ToString().ToLower(CultureInfo.CurrentCulture)}{targetName[1..]}";
+        }
+
+        private static async Task<Document> GetUpdatedDocumentForParameterRenameAsync(Document document, ISymbol parameter, string newName, CancellationToken cancellationToken)
```

Comment:
Because it duplicates a method of `ParameterNamesShouldMatchBaseDeclarationFixer`, it can be extracted to a shared internal helper method. But I'm not completely sure if it follows repository contribution guidelines (do not make changes to unrelated analyzers).